### PR TITLE
Add a CONTRIBUTING.md file

### PR DESCRIPTION
**Pull request by sunfishcode**
Here's an attempt at a CONTRIBUTING.md file, to welcome newcomers and answer questions about how to get involved in the project. I've included answers to questions that I get a lot, but if there are things I've missed that would help people get started, please bring them up!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+## Coding Guidelines
+
+For the most part, Cranelift follows common Rust conventions, though we do
+have a few additional things to be aware of.
+
+### rustfmt
+
+All PRs must be formatted according to the current stable rustfmt, and this is
+checked in the continuous integration tests. See
+[here](https://github.com/rust-lang-nursery/rustfmt#quick-start) for setup.
```
Comment (bnjbvr) [inline CONTRIBUTING.md]: Is it a good place to mention the `format-all.sh` script?

Comment (sunfishcode) [reply CONTRIBUTING.md]: Mentioned.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+missing or that don't make sense, or even that just don't work the way you
+expect them to, we're interested to hear about it!
+
+### Mentoring
+
+We're happy to mentor people, whether you're learning Rust, learning about
+compiler backends, learning about machine code, learning about how Cranelift
+does things, or all together at once.
+
+We tag issues in the issue tracker marked "good first issue" when we can, so
```
[the remark this answers is not in the document]
Comment (sunfishcode) [reply CONTRIBUTING.md]: Good idea.

**Comment by sunfishcode** [comment]
> mention ways to get in touch with other contributors, that is, give links to IRC / gitter rooms, or anything (it is probably somewhere else already, but I couldn't find it).

Added.

> not sure whether Cranelift has a Code of Conduct, but if it were the case, it would be nice to mention it here.

We don't have one yet, though I expect to propose one soon, at which point, yes, we should mention it in CONTRIBUTING.md.

> I'd add something about the process to contribute code: does one need to open an issue all the time? Or should PRs just be sent without previous discussion/approval? Are there specific commit message restrictions/recommendations? Is there a "one feature per commit" policy? Should every single commit compile and pass tests (which is practical for bisection purposes)?

> random idea: a word on governance might or might not be appropriate: what's the process to decide whether a PR is accepted or not? i.e. who gets the last word to decide whether a PR is accepted? (it's rather clear in my head, but might be nice to make it explicit)

I've now added a Development Process section which I think covers all these questions. Let me know what you think!

> Are there ways non technical contributions can be made to the project? (documentation for instance, talking about the project, etc. -- makes me wonder whether a list of public talks/slidedecks about Cranelift could find a way somewhere in the repository)

We don't have a lot right now, but it'd be cool to start collecting them.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -74,3 +74,36 @@ Our Python code is checked with [mypy](http://mypy-lang.org/) and
...
+Anyone may submit a pull request, and anyone may comment on or review others'
+pull requests. Pull requests are merged by members of the [Core Team].
+
+Core Team members may push minor changes directly, though should create PRs
+for significant changes.
+
+[issues]: https://guides.github.com/features/issues/
+[pull requests]: https://help.github.com/articles/about-pull-requests/
+[issue keywords]: https://help.github.com/articles/closing-issues-using-keywords/
+[Core Team]: https://github.com/orgs/CraneStation/teams/core-team
```
[the remark this answers is not in the document]
Comment (6A) [reply CONTRIBUTING.md]: I think `/teams` and `/teams/core-team` are restricted to members of the CraneStation organization. I think https://github.com/orgs/CraneStation/people would do fine though.

[the remark this answers is not in the document]
Comment (sunfishcode) [reply CONTRIBUTING.md]: Ah, thanks for pointing that out. I've updated it to the people link.

**Comment by sunfishcode** [comment]
Thanks for the feedback everyone! I expect this will continue to evolve, but it's a starting point :-).

**Pull request merged**